app/controllers/api/base_controller.rb

```diff
@@ -1,3 +1,27 @@
 class Api::BaseController < ApplicationController
   layout nil
-end
+
+  # POST /api/validate_email
+  def validate_email
+    email = params[:email]
+    if email_valid?(email)
+      if User.email_exists?(email)
+        AuthenticationEvent.new.log_email_validation(email, 'email_validation', current_user&.id)
+        render json: { accountExists: true, nextStep: 'authentication' }
```

Comment:
Returning detailed error messages like `email_not_found` can provide attackers with information about valid email addresses. Consider using a generic error message to avoid disclosing such information.